data_handler/handlers/loan_states/hashtack_v1/events.py

```diff
@@ -390,6 +397,56 @@ def __init__(
     # TODO: Reduce most of the events processing to `rewrite_original_collateral`, `rewrite_borrowed_collateral`, and 
     # `rewrite_debt`?
 
+    def process_updated_supply_token_price_event(self, event: pd.Series) -> None:
+        """
+        Example of transaction: https://starkscan.co/tx/0x028050442976bffce8d858a759f195e206c2d4424f93479b83665094236712f8#events
+        data column structure:
+           - 0 - token_supply # doesn't work for us
+           - 1 - underlying_asset # is applicable for `get_symbol`
+           - 2 - total_supply
+           - 4 - total_assets
+           - 6 - timestamp
+        :param event: pd.Series with event data
+        :return: None
+        """
+        token = get_symbol(event["data"][1])
+        # Convert total_supply and total_assets from hex to Decimal
+        total_supply = decimal.Decimal(str(int(event["data"][2], base=16)))
+        total_assets = decimal.Decimal(str(int(event["data"][4], base=16)))
+
+        # Calculate the cumulative interest rate
+        cumulative_interest_rate = total_assets / total_supply
+
+        # Store the interest rate in the collateral_interest_rate_models attribute
+
+        self.collateral_interest_rate_models.values[token] = cumulative_interest_rate
+
+    def process_updated_debt_token_price_event(self, event: pd.Series) -> None:
+        """
+        Example of transaction in starkscan: https://starkscan.co/event/0x0486fea45f212bba4600c959d3aca4108f442c10a3e2e4d2e97ccd8f2e59a834_5
+        Data structure of `event[data]`:
+            - debt_token: 0
+            - underlying_asset: 1
+            - total_supply: 2
+            - total_debt: 3
+            - timestamp: 4
+        :param event: pd.Series with event data
+        :return: None
+        """
+        token = get_symbol(event["data"][1])
+        # Convert total_supply and total_assets from hex to Decimal
+        total_supply = decimal.Decimal(str(int(event["data"][2], base=16)))
+        total_debt = decimal.Decimal(str(int(event["data"][4], base=16)))
```

Comment:
in docstring it says` total_debt: 3`. Please double check what is correct index to use here.

Reply:
yeah, good catch, it should be 4, thanks